pkg/storegateway/indexheader/reader_pool_test.go

```diff
@@ -170,14 +192,11 @@ func TestReaderPool_PersistLazyLoadedBlock(t *testing.T) {
 	snapshot.HeaderLastUsedTime = pool.LoadedBlocks()
 	err = snapshot.persist(tmpDir)
 	require.NoError(t, err)
-	require.NotContains(t, snapshot.HeaderLastUsedTime, blockID.String(), "lazyLoadedBlocks snapshot must be unset")
 
 	persistedData, err = os.ReadFile(persistedFile)
 	require.NoError(t, err)
-	expectedSnapshot = &lazyLoadedHeadersSnapshot{}
-	err = json.Unmarshal(persistedData, expectedSnapshot)
-	require.NoError(t, err)
-	require.Equal(t, 0, len(expectedSnapshot.HeaderLastUsedTime), "should have none of lazyLoaded index header")
+
+	require.Equal(t, `{"header_last_used_time":{},"user_id":"anonymous"}`, string(persistedData), "header_last_used_time should be cleared")
```

Comment:
Same as above - `JSONEq` is a better choice:
```suggestion
	require.JSONEq(t, `{"header_last_used_time":{},"user_id":"anonymous"}`, string(persistedData), "header_last_used_time should be cleared")
```